### PR TITLE
feat(server/bullmq): remove all finalized jobs by default

### LINKED ISSUE #2996: Review BullMQ cache persistence settings
See: https://docs.bullmq.io/guide/queues/auto-removal-of-jobs

### PR DESCRIPTION
**Pull request by ThatOneBro**
## What this PR does
* Adds removal of all finalized jobs (completed and failed) from Redis cache by default.
* Adds `bullmq` to config, so that users can configure `removeOnComplete` and `removeOnFail`
* Closes #2996

### CODE REVIEW

File: packages/server/src/workers/cron.ts [hunk 1]
```diff
@@ -41,11 +41,12 @@ let worker: Worker<CronJobData> | undefined = undefined;
  * Initializes the Cron worker.
  * Sets up the BullMQ job queue.
  * Sets up the BullMQ worker.
- * @param config The Redis config.
+ * @param redisConfig The Redis config.
+ * @param bullmqConfig The BullMQ config.
  */
-export function initCronWorker(config: MedplumRedisConfig): void {
+export function initCronWorker(redisConfig: MedplumRedisConfig, bullmqConfig?: MedplumBullmqConfig): void {
```
Comment (codyebberson) [inline packages/server/src/workers/cron.ts:47]: Thoughts on changing this to `config: MedplumServerConfig`, which would include both `redisConfig` and `bullmqConfig`?

File: packages/server/src/workers/cron.ts [hunk 2]
```diff
@@ -58,15 +59,19 @@ export function initCronWorker(config: MedplumRedisConfig): void {
...
+  });
   worker.on('completed', (job) => globalLogger.info(`Completed job ${job.id} successfully`));
   worker.on('failed', (job, err) => globalLogger.info(`Failed job ${job?.id} with ${err}`));
 }
 
 /**
  * Shuts down the Cron worker.
  * Closes the BullMQ job queue.
- * Clsoes the BullMQ worker.
+ * Closes the BullMQ worker.
```
Comment (codyebberson) [inline packages/server/src/workers/cron.ts:74]: Nice

File: packages/server/src/workers/cron.ts [hunk 3]
```diff
@@ -81,7 +86,7 @@ export async function closeCronWorker(): Promise<void> {
 }
 
 /**
- * Returns theCron queue instance.
+ * Returns the Cron queue instance.
```
Comment (codyebberson) [inline packages/server/src/workers/cron.ts:89]: Nice

**Review by codyebberson — APPROVED** [review]
Overall, this looks good.  One minor API question, not blocking.

File: packages/server/src/workers/cron.ts (hunk 1 above)
Comment (ThatOneBro) [reply packages/server/src/workers/cron.ts:47]: Yeah I almost did that, took like 10 minutes trying to decide between the two signatures.

I think we should do that now that I got your approval 👍

**Pull request merged**